c/compare.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python3
+
+# This script iterates over all tasks and performs the following check:
+# For each file with existing ".c"- and ".i"-file
+# we generate the goto-cc program (i.e., intermediate representation of cbmc)
+# and compare the goto-cc programs with goto-diff.
+# Finding a difference implies a wrong preprocessing.
+
+import glob
+import yaml
+import os
+import shutil
+import subprocess
+from optparse import OptionParser
```

Comment:
Note that `optparse` is deprecated since 2011, this is mentioned in the first line of its documentation.